activerecord/lib/active_record/persistence.rb

```diff
@@ -428,6 +428,14 @@ def new_record?
       @new_record
     end
 
+    # Returns true if this object was just created -- that is, prior to the last
+    # save, the object didn't exist in the database and new_record? would have
+    # returned true.
+    def previously_new_record?
+      sync_with_transaction_state
```

Comment:
Can you sync the line with one above and below methods?

```suggestion
      sync_with_transaction_state if @transaction_state&.finalized?
```

https://github.com/rails/rails/blob/251037efb383c7cfeba98cdc7e6fc8a7f12933a9/activerecord/lib/active_record/persistence.rb#L426-L427
https://github.com/rails/rails/blob/251037efb383c7cfeba98cdc7e6fc8a7f12933a9/activerecord/lib/active_record/persistence.rb#L440-L441
https://github.com/rails/rails/blob/251037efb383c7cfeba98cdc7e6fc8a7f12933a9/activerecord/lib/active_record/persistence.rb#L447-L448